the-collab-lab/tcl-69-smart-shopping-list
# [MF] Refactoring for readability and maintenance

## Description

This code extracts most of the logic from List.jsx that handles sorting of the list items by urgency and alphabetically. It creates new files to handle this abstraction. There is a new utils file that will hold some of the sorting and filtering functions. These functions are then imported into the List.jsx and a new SortedDataMap component. Once these functions have run the SortedDataMap will render the sorted list items onto the current /list page.

In addition the inputs in the `add item` dialog have been removed from List.jsx and pasted into the new AddItem component with the handleInput change functionality to update the form state values. 

The goal was to trim the size of the file and better organize for maintainability.

## Related Issue

closes #72 
subissue #14 

## Acceptance Criteria
- [x] Create utils file that handles sorted logic
- [x] Create component for entering new Item 
- [x] Separate the sorting data jsx into its own component
- [x] Use the `sortedData` utils file to effectively filtered data and sort into urgency arrays

## Type of Changes

Use one or more labels to help your team understand the nature of the change(s) you’re proposing. E.g., `bug fix` or `enhancement` are common ones.

## Updates

Does not update the UI

## Testing Steps / QA Criteria

Upon login the app should function in the same way prior to refactor. Try adding an item and verifying that item is sorted correctly.
